src/platform/graphics/texture.js

```diff
@@ -926,6 +883,40 @@ class Texture {
         this.impl.uploadImmediate?.(this.device, this);
     }
 
+    /**
+     * Download texture's top level data from graphics memory to local memory.
+     *
+     * @ignore
+     */
+    async downloadAsync() {
+        const promises = [];
+        for (let i = 0; i < (this.cubemap ? 6 : 1); i++) {
+            const renderTarget = new RenderTarget({
```

Comment:
I wonder if we can move more of this code to the device, to avoid having to import render target module into the texture, which sounds like a future circular reference.

Also, WebGPU will likely not need to create it.

Reply:
This is probably ok for now since it's private API?

WebglGraphicsDevice already supported `readPixel` which WebGpu doesn't. So this function and those will need updating when you add WebGpu support.

So yeah, I'll be happy to leave this till you do the deeper refactor.